File: 프로그래머스/1/완주하지 못한 선수/완주하지못한선수.js
function solution(participant, completion) {
  let hashed = [];
  participant.forEach((entry) => {
    hashed[entry] = hashed[entry] ? hashed[entry] + 1 : 1;
  });

  completion.forEach((entry) => {
    hashed[entry] = hashed[entry] - 1;
  });

  console.log(hashed);

  for (var key in hashed) {
    if (hashed[key] >= 1) return key;
  }
}

solution(
  ["marina", "josipa", "nikola", "vinko", "filipa"],
  ["josipa", "filipa", "marina", "nikola"]
);

File: 프로그래머스/1/완주하지 못한 선수/완주하지못한선수.py
def solution(participant, completion):
    if len(completion) == 0:
        return participant[0]
    sortParti = sorted(participant)
    sortCompl = sorted(completion)

    for i in range(len(sortParti)):
        if sortParti[i] != sortCompl[i] or len(sortParti) == i:
            return sortParti[i]

    
    
    
print(solution(["mislav", "stanko", "mislav", "ana"], ["stanko", "ana", "mislav"]))
